src/transformation/visitors/language-extensions/multi.ts

```diff
@@ -1,169 +1,92 @@
     const type = context.checker.getTypeAtLocation(expression.expression);
     return type.symbol?.declarations?.some(isMultiFunctionDeclaration) ?? false;
 }
 
 export function returnsMultiType(context: TransformationContext, node: ts.CallExpression): boolean {
     const signature = context.checker.getResolvedSignature(node);
-    return signature?.getReturnType().aliasSymbol?.declarations?.some(isMultiTypeDeclaration) ?? false;
+    const type = signature?.getReturnType();
+    return type ? isMultiReturnType(type) : false;
+}
+
+export function isMultiReturnCall(context: TransformationContext, expression: ts.Expression) {
+    return ts.isCallExpression(expression) && returnsMultiType(context, expression);
 }
 
 export function isMultiFunctionNode(context: TransformationContext, node: ts.Node): boolean {
     const type = context.checker.getTypeAtLocation(node);
     return type.symbol?.declarations?.some(isMultiFunctionDeclaration) ?? false;
 }
 
-export function transformMultiCallExpressionToReturnStatement(
-    context: TransformationContext,
-    expression: ts.Expression
-): lua.Statement {
-    assert(ts.isCallExpression(expression));
-
-    const expressions = transformArguments(context, expression.arguments);
-    return lua.createReturnStatement(expressions, expression);
-}
-
-export function transformMultiReturnStatement(
-    context: TransformationContext,
-    statement: ts.ReturnStatement
-): lua.Statement {
-    assert(statement.expression);
-
-    return transformMultiCallExpressionToReturnStatement(context, statement.expression);
-}
-
-function transformMultiFunctionArguments(
-    context: TransformationContext,
-    expression: ts.CallExpression
-): lua.Expression[] | lua.Expression {
-    if (!isMultiFunction(context, expression)) {
-        return context.transformExpression(expression);
-    }
-
-    if (expression.arguments.length === 0) {
-        return lua.createNilLiteral(expression);
+export function isInMultiReturnFunction(context: TransformationContext, node: ts.Node) {
+    const declaration = findFirstNodeAbove(node, ts.isFunctionLike);
+    if (!declaration) {
+        return false;
     }
-
-    return expression.arguments.map(e => context.transformExpression(e));
+    const signature = context.checker.getSignatureFromDeclaration(declaration);
+    const type = signature?.getReturnType();
+    return type ? isMultiReturnType(type) : false;
 }
 
-export function transformMultiVariableDeclaration(
-    context: TransformationContext,
-    declaration: ts.VariableDeclaration
-): lua.Statement[] {
-    assert(declaration.initializer);
-    assert(ts.isCallExpression(declaration.initializer));
-
-    if (!ts.isArrayBindingPattern(declaration.name)) {
-        context.diagnostics.push(invalidMultiTypeToNonArrayBindingPattern(declaration.name));
-        return [];
+export function multiReturnCallShouldBeWrapped(context: TransformationContext, node: ts.CallExpression) {
```

Comment:
I would prefer
```suggestion
export function shouldMultiReturnCallBeWrapped(context: TransformationContext, node: ts.CallExpression) {
```

But I am also okay with the current version.